pandas/core/index.py

```diff
@@ -1203,11 +1203,9 @@ def take(self, indexer, axis=0):
         numpy.ndarray.take
         """
 
-        indexer = com._ensure_platform_int(indexer)
-        taken = np.array(self).take(indexer)
-
-        # by definition cannot propogate freq
-        return self._shallow_copy(taken, freq=None)
+        indices = com._ensure_platform_int(indices)
+        taken = np.array(self).take(indices)
```

Comment:
I realize that this is actually a perf issue coming from here
do the take directly in self.values as take always copies anyhow
(this may fail for categoricla) and u might be able to get away with np.array(arr, copy=False)